src/Aspire.Hosting/Dcp/DcpExecutor.cs

```diff
@@ -994,8 +974,29 @@ private async Task CreateExecutableAsync(AppResource er, ILogger resourceLogger,
         {
             spec.Args.AddRange(projectArgs);
         }
-        spec.Args.AddRange(args.Select(a => a.Value));
-        er.DcpResource.SetAnnotationAsObjectList(CustomResource.ResourceAppArgsAnnotation, args.Select(a => new AppLaunchArgumentAnnotation(a.Value, isSensitive: a.IsSensitive)));
+
+        var launchArgs = new List<(string Value, bool IsSensitive)>();
+
+        // If the executable is a project then include any command line args from the launch profile.
+        if (er.ModelResource is ProjectResource project)
```

Comment:
We have to make sure this does not change any behavior when there's IDE in use. In that situation the args is handled by the IDE even if we show them in the dashboard.

Reply:
I tested when running TestShop from IDE. Should I check I get the same behavior when running form the command line? I don't understand the scenario you're talking about.

Reply:
The code before has 2 branches, in IDE mode command line lanuch profile (and command line args) are processed 
by the IDE https://github.com/dotnet/aspire/blob/e574625ccf3f8c126e9822ee5015c6434c09ba87/src/Aspire.Hosting/Dcp/DcpExecutor.cs#L863-L872.

In your change that is happening in both the IDE and non IDE case regardless. We can process the launch profile for args, but if we want the behavior to match today, we shouldn't provide the args to the DCP resource (in this one case).

If we change this (which I am not 100% against), the IDE will never set arguments from the launch profile when doing processing. This means all of the weird VS tokens you can put in launch settings that only work in VS will stop working.

Reply:
As a test, put $(SolutionDir) in the args and run before and after your change and see if it gets resolved

Reply:
Ok, I'll look at this more tomorrow. If we do change something, I think it should probably be a 9.2 change. Give us time to see if it causes any issues.